wrappers/_shared.nix

```diff
@@ -1,37 +1,66 @@
-helpers:
-{ lib, config, ... }:
+{
+  filesOpt ? null,
+  filesPrefix ? "nvim/",
+  initName ? "init.lua",
+}:
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
 let
   inherit (lib)
     isAttrs
     listToAttrs
     map
+    mkIf
+    mkMerge
     mkOption
     mkOptionType
+    optionalAttrs
+    setAttrByPath
     ;
   cfg = config.programs.nixvim;
   extraFiles = lib.filter (file: file.enable) (lib.attrValues cfg.extraFiles);
 in
 {
-  helpers = mkOption {
-    type = mkOptionType {
-      name = "helpers";
-      description = "Helpers that can be used when writing nixvim configs";
-      check = isAttrs;
-    };
-    description = "Use this option to access the helpers";
-    default = helpers;
-  };
+  imports = [
+    (lib.mkRenamedOptionModule
+      [
+        "nixvim"
+        "helpers"
+      ]
+      [
+        "lib"
+        "nixvim"
+      ]
+    )
+  ];
 
-  # extraFiles, but nested under "nvim/" for use in etc/xdg config
-  configFiles = listToAttrs (
-    map (
-      { target, source, ... }:
-      {
-        name = "nvim/" + target;
-        value = {
-          inherit source;
-        };
-      }
-    ) extraFiles
-  );
+  config = mkMerge [
+    # Add our lib to `config.lib`
+    { lib.nixvim = import ../lib/helpers.nix { inherit pkgs lib; }; }
```

Comment:
Do we want to just use `lib/helpers.nix`, or the entire `lib/default.nix` (which includes `helpers` as an attr)?

I guess functions outside of `helpers` may be useful in some cases, or useful ones may be added in the future?
It is also more consistent with the flake `outputs.lib.<system>`.

Reply:
That's a good point.
Would it break a user config ? I mean, would the path of the existing helper functions change ?

Reply:
> Would it break a user config ? I mean, would the path of the existing helper functions change ?

Long-term, having what is now `helpers` just be a subset of our nixvim lib, may be better. You'd end up with `lib.nixvim.checks`, `lib.nixvim.options`, `lib.nixvim.types`, with a few top-level aliases.

As for now, we can either have `lib.nixvim.{checks,helpers}` or we can do something like:

```nix
# Copy .helpers to the lib's: top-level
let
  nixvimLib = import ../lib { inherit lib pkgs; };
in
lib.nixvim = nixvimLib // nixvimLib.helpers;
```

Either way, we can make it non-breaking with the `mkRenamedOptionModule` option. We either rename to `lib.nixvim` or `lib.nixvim.helpers`, depending on our long-term plan.

Reply:
If preferred, I can drop the rename to `config.lib.*` from this PR and tackle that once we have a firmer roadmap for our lib & helpers?

Reply:
Yes, this would need further discussion.

Reply:
I think we all agree it's better to use the `lib` option provided by host modules.

So the reason to wait is to decide exactly how we want to use the option and attempt to get it right first try?

Reply:
I've dropped the option rename from the PR now